pkg/jx/cmd/clients/factory.go

```diff
@@ -382,13 +383,45 @@ func (f *factory) ResetSecretsLocation() {
 
 // CreateSystemVaultClient gets the system vault client for managing the secrets
 func (f *factory) CreateSystemVaultClient(namespace string) (vault.Client, error) {
-	name, err := kubevault.SystemVaultName(f.kubeConfig)
+	name, err := f.getVaultName(namespace)
 	if err != nil {
-		return nil, errors.Wrap(err, "building the system vault name from cluster name")
+		return nil, err
 	}
 	return f.CreateVaultClient(name, namespace)
 }
 
+func (f *factory) getVaultName(namespace string) (string, error) {
+	name := ""
+	context, err := cluster.Context(f.kubeConfig)
+	if err != nil {
+		return name, err
+	}
+	if context == nil {
+		kubeClient, _, err := f.CreateKubeClient()
+		if err != nil {
+			return name, err
+		}
+		cm, err := kube.GetConfigMap(kubeClient, namespace, kube.ConfigMapNameJXInstallConfig)
+		if err != nil {
+			return name, errors.Wrapf(err, "no ConfigMap %s in namespace %s", kube.ConfigMapNameJXInstallConfig, namespace)
+		}
+		if cm.Data == nil {
+			return name, fmt.Errorf("no data for ConfigMap %s in namespace %s", kube.ConfigMapNameJXInstallConfig, namespace)
+		}
+		name = cm.Data["clusterName"]
```

Comment:
This const is defined in `kube.ClusterName`.

Reply:
lemme use that